homework3/src/main/java/homework4/MainHobby.java

```diff
@@ -0,0 +1,60 @@
+package main.java.homework4;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class MainHobby {
+    public static void main(String[] args) {
+        PracticeHobby r = new PracticeHobby();
+
+        System.out.println("The first person with his hobbies: ");
+        r.put(new Person("Alin Tudor", 15), Arrays.asList(
+                new Hobby("Football", 2, Arrays.asList(new Adress("Cluj", "Aleea Stadionului", "2"),
+                        new Adress("Cluj", "Drumul Sfântul Ioan", "3"))),
+                new Hobby("Dance", 1, Arrays.asList(new Adress("Cluj", "Strada Someșului", "5"),
+                        new Adress("Cluj", "Regele Ferdinand", "22")))));
+        r.print();
+        System.out.println();
+
+        //adding an existing  Person with the same properties
+        System.out.println("Adding an existing  Person with the same properties: ");
+        r.put(new Person("Alin Tudor", 15), Arrays.asList(
```

Comment:
destul de complicat initializarea asta. In cazurile astea poti face obiecte indepedente si apoi le aduni in acea lista. E mai usor de urmarit.